src/test/java/BaseTest.java

```diff
@@ -1,18 +1,142 @@
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.interactions.Action;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
 import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Parameters;
+
+import java.time.Duration;
 
 
 public class BaseTest {
+   public static WebDriver driver = null;
+   public static String url = null;
+   public static WebDriverWait wait = null;
 
+   public static Actions actions = null;
+  
     @BeforeSuite
     static void setupClass() {
         WebDriverManager.chromedriver().setup();
     }
+    @BeforeMethod
+    @Parameters({"baseUrl"})
+    public void launchBrowser(String baseUrl){
+        ChromeOptions options = new ChromeOptions();
+        options.addArguments("--disable-notifications","--remote-allow-origins=*","--incognito","--start-maximize");
+
+        driver = new ChromeDriver(options);
+        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+        driver.manage().window().maximize();
+        url = baseUrl;
+        driver.get(url);
+        
+    }
+
+    public static void provideEmail(String email){
+     WebElement emailField = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("[type='email']")));
+     emailField.click();
+     emailField.sendKeys(email);
+     
+    }
+      public static void providePassword(String password){
+        WebElement passwordField = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("[type='password']")));
+        passwordField.click();
+        passwordField.sendKeys(password);
+      }
+    
+      public static void clickSubmit(){
+        WebElement submitButton = wait.until(ExpectedConditions.elementToBeClickable
+                (By.cssSelector("[type='submit']")));
+        submitButton.click();
+  
+      }
+      public static void logIn(String email, String password){
+        provideEmail(email);
+        providePassword(password);
+        clickSubmit();
+        
+      }
+        public static void createNewPlaylist(){
+        WebElement newPlaylistIcon = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("i[title='create new playlist']")));
+        newPlaylistIcon.click();
+        WebElement newPlayList = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("li[data-testid='playlist-context-menu-create-simple']")));
+        newPlayList.click();
+        WebElement nameField = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("input[name='name']")));
+        nameField.clear();
+        nameField.sendKeys("Pluto");
+        nameField.sendKeys(Keys.ENTER);
+        }
+
+        public static boolean newPlaylistIsDisplayed(){
+        WebElement newPlaylist = wait.until(ExpectedConditions.presenceOfElementLocated(By.cssSelector("a[class='active']")));
```

Comment:
Please finish the homework!You got this problem, please correct it. 

![image](https://user-images.githubusercontent.com/104761482/225847057-a016742b-b75f-4527-aeb4-ab7634d1f7d0.png)

HINT


![Test status](https://user-images.githubusercontent.com/104761482/225847164-2ab95efc-e115-4cdb-abec-3796e979816f.png)


![Test error_](https://user-images.githubusercontent.com/104761482/225847199-dc773ae1-fe62-49fe-85bd-58b226da6d5a.png)